### PR TITLE
Psammead to Simorgh Component Migration [WSTEAMA-51]

### PR DESCRIPTION
**Pull request by pvaliani**
Resolves Jira Ticket WSTEAMA-51

**Overall change:**
- Migrates all Psammead components (unless deleted where stated) except for `@bbc/web-vitals` to a module called `legacy` in Simorgh
- Psammead components cannot be viewed in Simorgh storybook (at the moment) but Simorgh Storybook still works

**Code changes:**

- Removes `changelogs` and `package.json` files from each migrated component in `#legacy` module
- Removes `@bbc/psammead` `package.json` dependency entries from Simorgh's `package.json`
- Adds the following dependencies to Simorgh `package.json`:
  - `jalaali-js`
  - `xmldoc`
  - `octokit/rest`
  - `octokit/action`
  - `find-cache-dir`
  - `juggle/resize-observer`
  - `mkdirp`
  - `json5`
- Downgrades `moment` to  v2.24.0 and `moment-timezone` to v0.5.26 in Simorgh for compatibility with imported Psammead components
- Adds `babel/plugin-proposal-export-default-from` plugin and updates module exports to default export type to resolve component imports
- Removes irrelevent and unnecessary unit tests relating to the import/export of now unused `package.json` files within migrated psammead components
- Adds a `#legacy` alias to `dirAlias.js`, `jsconfig.json` and webpack config
- Adds a `watchOptions` entry with `ignored` using glob pattern to the `webpack config` which ignore the `tz` `moment-timezone-include` files. This caused webpack to continuously loop upon running the development server with `yarn dev`
- Replaces Psammead dependencies in `service configs` with absolute paths due to legacy folders due to lack of aliasing support 
- Removes unused `psammead-oembed` component and related `dompurify` dependency from Simorgh `package.json`
- Adds a glob pattern to `storybook` `main.js` to path to `legacy` so that Simorgh storybook still builds and functions correctly
- Updates snapshots to reflect new classnames generated for html elements due to migration of the components
- Updates `escapeRegex` within `psammead-media-player` to reflect Simorgh's `media-player`. Prevents regEx escapes on our domain and enables our GH actions CodeQL checks to pass

**Steps**:

 ✅ - **Completed**
❌ - **Not Completed**

Process *BEFORE* the start of the migration was set out and is described below:

- Copy all components from Psammead across to Simorgh into a 'legacy' folder in Simorgh to differentiate them from components we will move to in the future that meet our 'component standards' ✅
- Perform a mass find and replace to remove references to Psammead packages from Simorgh package.json - all "@bbc/psammead/ references, @bbc/gel-foundations replace with a #legacy alias ✅

- Then for each package we will bring it in line with how components are structured in Simorgh. We need to convert the packages from separate NPM packages to local components in Simorgh, this involved:
    - Deleting the package.json from each psammead component
    - hoisting any dependencies up to the root package.json in Simorgh ✅
    - Deleting changelog.md ✅
    - Deleting dist, esm, node_modules for individual package build artefacts (either do a fresh clone of Psammead without them or remove them from the legacy folder when copying over). ✅
Note: Any Psammead components that have NPM dependencies need to have their dependencies added to Simorgh package.json. For example:
https://github.com/bbc/psammead/blob/latest/packages/containers/psammead-timestamp-container/package.json which requires "moment-time". Hoist these dependencies to the top level of the Simorgh package.json ✅

- Consider moving content of src into root (move all source code up one level within the Psammead component folder name), and delete the now empty src directory - this may be essential, see if it breaks first ❌
- Check that the Psammead component tests run correctly - may need to update Simorgh test script ✅

**Additional work to be completed before merging this PR**:
- Investigate and reduce bundle sizes down to acceptable threshold. Largest bundle size is currently ~1300kB (https://github.com/bbc/simorgh/pull/10110) ✅ (see below)
- Perform change freeze (and probably merge freeze) and do a broad regression test across the site before putting live
Example (spike) PR: https://github.com/bbc/simorgh/pull/9966

**Additional work to be completed related to this PR at a future date**:
- Migrate `@bbc/web-vitals` from Psammead [WSTEAMA-79]
- Fix Psammead components to be viewable in Simorgh storybook [[NEWSWORLDSERVICE-1534]
- Decide the future of Psammead NPM packages and the Psammead repo [WSTEAMA-80]
- Upgrade Moment [NEWSWORLDSERVICE-1533] (https://github.com/bbc/simorgh/pull/10128) ✅

**BUNDLE SIZE FIX**

The Bundle Sizes have been addressed in this PR:

https://github.com/bbc/simorgh/pull/10110 

The largest bundle size is currently ~1090kB but further work should be completed to reduce this where possible

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by DarioR01** [comment]
I might have accidentally changed the test logic of psammead-locales... All the files can be identified by running yarn test and looking for error 'i' is already declared in the upper scope on line 16 column 7

I approached this by removing the eslintrc file in psammead locales, but we could also install esling-plugin-es5 so that psammead-locales uses es5 instead.

**Comment by paruchurisilpa** [comment]
Testing:

Have regression tested links for few services on this document below locally -
https://github.com/bbc/simorgh/blob/latest/docs/Simorgh-Release-Info.md

Covered:
Chrome Latest
Safari on Last 2 IOS versions
Safari 13.x
Safari 14.x
Opera Mini (Android)
Including Extreme Mode
Latest Opera on Desktop
Latest Firefox on Desktop

**Comment by paruchurisilpa** [comment]
Cypress tests seem to have passed too. Happy for this to be merged..

File: .eslintrc.js [hunk 1]
```diff
@@ -21,7 +21,8 @@ module.exports = {
       jsx: true,
     },
   },
-  plugins: ['prettier', 'json', 'jsx-a11y', 'react-hooks', 'cypress'],
+  ignorePatterns: ['**/tz/**', 'index.stories.jsx', 'index.amp.stories.jsx'],
```
Comment (MuhammadAHussain) [inline .eslintrc.js:24]: Curious why we are ignoring story pages now?

File: src/app/containers/ArticleTimestamp/timeFormatTests/expectedFormats.json [hunk 2]
```diff
@@ -101,10 +101,10 @@
   },
   "gujarati": {
     "default": {
-      "one minute ago": "એક મિનિટ પહેલા",
-      "five minutes ago": "5 મિનિટ પહેલા",
-      "one hour ago": "એક કલાક પહેલા",
-      "five hours ago": "5 કલાક પહેલા",
```
Comment (amoore108) [inline src/app/containers/ArticleTimestamp/timeFormatTests/expectedFormats.json:107]: Looks like a slight difference in formatting here. May not be an issue as it seems to only affect this service.

File: .eslintrc.js (hunk 1 above)
Comment (pvaliani) [reply .eslintrc.js:24]: It's to do with the linting errors we are receiving at the moment due to Psammead components not being viewable in Simorgh. There are currently two separate `main.js` config files between the Simorgh and Psammead repo's which are used to configure their respective storybooks. A ticket has been setup and is detailed in the body of this PR description in order to address work required to get Psammead components viewable in Simorgh.

File: src/app/containers/ArticleTimestamp/timeFormatTests/expectedFormats.json (hunk 2 above)
Comment (pvaliani) [reply src/app/containers/ArticleTimestamp/timeFormatTests/expectedFormats.json:107]: Can keep an eye on it! However, there's a PR and ticket in the body description to address the downgrading in this PR of Moment (which may be the culprit) and to re-upgrade it post-deployment which might solve any formatting discrepancies.

**Review by amoore108 — APPROVED** [review]
Looks good. Great work by everyone involved! 🚀

**Review by MuhammadAHussain — APPROVED** [review]
LGTM 👍

**Pull request merged**